Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md

```diff
@@ -0,0 +1,88 @@
+# OOP
+------------
+## Contents
+1. Introduction to class and object 
+2. What is a class ?
+3. What is an object?
+4. Difference between Classes and objects in Java
+5. Memory allocation 
+ 
+-------------
+## Introduction to class and object 
+- Classes and objects are one of the most fundamental units of many Java application and frameworks. 
+- As Java is an object oriented programming language we can see classes and objects everywhere. So it's quite important to understand these concepts .
+- If we take example of our surrounding we can find objects everywhere like car, dogs, humans etc .
+- All these thing have something in common and that is 
+    1. **States**
+    2. **Behaviour**
+- And these states and behaviour are used to differentiate them properly.   
+    **FOR EXAMPLE :**
+    Humans can talk but dogs can't talk they bark. So here the talking of the human and barking of the dogs are their behaviour and these behaviour help us to differentiate between objects , in this case humans and dogs. So here you might be having some confusion about states and behaviour. 
+- **State** is nothing but the values or the data . In human case their Name and gender are considered as state .
+- **Behaviour** is the action or the work or the task the object performs. Here human is able to talk, talking is action or we can say behaviour. Similarly barking is a behaviour of dog. For better understanding you can look at this image 
+<!-- Image stating the difference between states and behaviour -->
+![state and behaviour](https://github.com/amitx007/winter-of-contributing/blob/Java/Java/Fundamentals/Assets/IMG1.jpg)
+- Now let us talk about classes. In layman term **Classes** are nothing but the logical description of how an object will behave and what are the states the object is going to have.
+- So let us take another **Example**. When you are creating a house, what is the first thing you do ? Yes you do the planning and make a sketch of it and if we make this sketch  with proper measurements and all, then it will be a **Blueprint**. So Blueprint will contain all the details and measurements of the floor ,ceiling and how it will be designed .
+<!-- Image of a house (blueprint) -->
+![class as a blueprint](https://github.com/amitx007/winter-of-contributing/blob/Java/Java/Fundamentals/Assets/IMG2.jpg)
```

Comment:
You can use relative markdown as well. Change the URL to `../Assets/IMG2.jpg`.

Reply:
everything done but couldn't understand what u mean by points:?

Reply:
@AkMo3  can u tell me abt the point that u have mentioned i couldn't understand